source/GameData.cpp

```diff
@@ -333,9 +344,46 @@ void GameData::CheckReferences()
 		if(it.second.Name().empty())
 			Warn("phrase", it.first);
 	// Planet names are used by a number of classes.
-	for(auto &&it : planets)
-		if(it.second.TrueName().empty() && !NameIfDeferred(deferred["planet"], it))
-			NameAndWarn("planet", it);
+	{
+		vector<pair<const string, Planet>*> planetsThatAreActuallyWormholesMaybe;
+		for(auto &&it : planets)
+			if(it.second.Name().empty() && !NameIfDeferred(deferred["planet"], it))
+			{
+				if(!player.HasVisited(it.second))
+				{
+					// Hasn't been visited so can't be a wormhole.
+					NameAndWarn("planet", it);
+					continue;
+				}
+
+				// Ugly workaround to support old style visited planet that were
+				// actually wormholes not planets.
+				planetsThatAreActuallyWormholesMaybe.push_back(&it);
+			}
+
+		// Loop through every system to try to discover planets that are actually wormholes.
+		for(auto &&system : systems)
+			for(auto &&planet : planetsThatAreActuallyWormholesMaybe)
+			{
+				string name = planet->first + " (" + system.first + ")";
```

Comment:
It's not requiring it. It's how the new wormholes are named. Should I instead hardcode the existing wormhole names? Because I really don't see what else I can do.